source/extensions/http/original_ip_detection/xff/xff.cc

```diff
@@ -10,13 +11,43 @@ namespace Xff {
 
 XffIPDetection::XffIPDetection(
     const envoy::extensions::http::original_ip_detection::xff::v3::XffConfig& config)
-    : xff_num_trusted_hops_(config.xff_num_trusted_hops()) {}
+    : xff_num_trusted_hops_(config.xff_num_trusted_hops()),
+      recurse_(config.xff_trusted_cidrs().recurse()) {
+  xff_trusted_cidrs_.reserve(config.xff_trusted_cidrs().cidrs().size());
+  for (const envoy::config::core::v3::CidrRange& entry : config.xff_trusted_cidrs().cidrs()) {
+    Network::Address::CidrRange cidr = Network::Address::CidrRange::create(entry);
+    xff_trusted_cidrs_.push_back(cidr);
+  }
+}
 
 XffIPDetection::XffIPDetection(uint32_t xff_num_trusted_hops)
-    : xff_num_trusted_hops_(xff_num_trusted_hops) {}
+    : xff_num_trusted_hops_(xff_num_trusted_hops), recurse_(false) {}
+
+XffIPDetection::XffIPDetection(
```

Comment:
This constructor (rightly) causes a coverage failure. I'm not actually sure if it's needed.

Its sibling (directly above) is only used by some tests but not directly via HCM, so perhaps it's not needed?

Reply:
I've removed this constructor.